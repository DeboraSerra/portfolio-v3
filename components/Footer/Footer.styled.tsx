import styled from "styled-components";

export const Footer = styled.footer`
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  background-color: ${props => props.theme.bg.secondary};
  min-height: 64px;
  padding: 12px 24px;
  position: relative;
  bottom: 0;

  .footer {
    &__heading {
      font-size: ${props => props.theme.fontSizes[2]};
      color: ${props => props.theme.text.secondary};
    }// &__heading

    &__nav {
      display: flex;
      flex-direction: column;
      align-items: flex-start;
      justify-content: center;
      width: 30%;

      font-size: ${props => props.theme.fontSizes[0]};

      &--item {
        line-height: ${props => props.theme.lineHeights.body};
      }// &:a
    }// &__nav
  }// .footer

  @media only screen and (max-width : 576px) {
    flex-direction: column;
    gap: 16px;
    align-items: center;

    .footer {
      &__heading {
        text-align: center;
      }// &__heading
      
      &__nav {
        align-items: center;
        width: 100%;
      }// &__nav
    }// .footer
  }// @media only screen and (max-width : 576px)
`;
