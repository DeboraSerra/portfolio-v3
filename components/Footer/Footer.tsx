import * as S from "./Footer.styled";

const Footer = () => {
  return (
    <S.Footer>
      <div className="container">
        <h2 className="footer__heading">Developed by DÃ©bora Serra</h2>
        <nav className="footer__nav">
          <a
            target="_blank"
            rel="noreferrer"
            className="footer__nav--item"
            href="https://github.com/DeboraSerra"
          >
            My github page
          </a>
          <a
            target="_blank"
            rel="noreferrer"
            className="footer__nav--item"
            href="https://wa.me/+5561982311582"
          >
            My what&apos;s app
          </a>
          <a
            target="_blank"
            rel="noreferrer"
            className="footer__nav--item"
            href="https://www.linkedin.com/in/debora-r-serra/"
          >
            My Linked-in
          </a>
        </nav>
      </div>
    </S.Footer>
  );
};

export default Footer;
