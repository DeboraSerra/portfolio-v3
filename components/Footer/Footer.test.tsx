import { render } from "@testing-library/react";
import Footer from "./Footer";

describe("Test the Footer component", () => {
  it('Should match the snapshot', () => {
    const { asFragment } = render(<Footer />)
    expect(asFragment()).toMatchSnapshot();
  })
})