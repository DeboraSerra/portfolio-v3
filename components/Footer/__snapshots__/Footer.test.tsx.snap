// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Footer component Should match the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  background-color: #f1b2b2;
  min-height: 64px;
  padding: 12px 24px;
  position: relative;
  bottom: 0;
  box-shadow: 0 -12px 24px #45496a6f;
}

.c0 .footer__heading {
  font-size: 18px;
  color: #2f3037;
}

.c0 .footer__nav {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 30%;
  font-size: 14px;
}

.c0 .footer__nav--item {
  line-height: 1.5;
}

@media only screen and (max-width :576px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    gap: 16px;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c0 .footer__heading {
    text-align: center;
  }

  .c0 .footer__nav {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    width: 100%;
  }
}

<footer
    class="c0"
  >
    <h2
      class="footer__heading"
    >
      Developed by DÃ©bora Serra
    </h2>
    <nav
      class="footer__nav"
    >
      <a
        class="footer__nav--item"
        href="https://github.com/DeboraSerra"
        rel="noreferrer"
        target="_blank"
      >
        My github page
      </a>
      <a
        class="footer__nav--item"
        href="https://wa.me/+5561982311582"
        rel="noreferrer"
        target="_blank"
      >
        My what's app
      </a>
      <a
        class="footer__nav--item"
        href="https://www.linkedin.com/in/debora-r-serra/"
        rel="noreferrer"
        target="_blank"
      >
        My Linked-in
      </a>
    </nav>
  </footer>
</DocumentFragment>
`;
